### PR TITLE
Add support for using different modes for each phase of an xspi operation

### PR DESCRIPTION
**Pull request by dne**
This adds `Config.modes` and `Xspi.configure_modes` methods but keeps the existing API (which sets the same mode for all phases).

### CODE REVIEW

File: src/xspi/mod.rs
Comment (Abestanis) [inline src/xspi/mod.rs:191]:
```suggestion
        pub fn all(mode: XspiMode) -> Self {
```
Suggestion (I just saw this while browsing around, feel free to ignore): If this is called `all` then it is very clear that all modes are set to the given mode.

**Comment by richardeoin** [comment]
Nice that this can be added without breaking backwards compatibility! I think `configure_modes` should have some documentation ("use this if you want different modes for each phase..") but otherwise looks good.

If you're able to share, it would be great to know if you have tested this on hardware and what the results were.

**Comment by dne** [comment]
> Nice that this can be added without breaking backwards compatibility! I think `configure_modes` should have some documentation ("use this if you want different modes for each phase..") but otherwise looks good.

Thanks for reviewing! I've amended the PR to improve the docs, and also attempting to fix the octospi variant that failed in CI.

> If you're able to share, it would be great to know if you have tested this on hardware and what the results were.

I've used this successfully with the 'H745 and Infineon S25FL127S QSPI Flash (in quad mode), using the 4QOR and 4QPP commands ("quad output read" and "quad page program" with 4-byte addresses).

File: src/xspi/mod.rs
Comment (dne) [reply src/xspi/mod.rs:191]: > Suggestion (I just saw this while browsing around, feel free to ignore): If this is called `all` then it is very clear that all modes are set to the given mode.

@Abestanis Thanks for the suggestion; I'll let the maintainers decide what they prefer.

Comment (richardeoin) [reply src/xspi/mod.rs:191]: I see the reasoning behind calling it `all`, but `new` is definitely closer to the [naming convention](https://rust-lang.github.io/api-guidelines/naming.html#c-case) and is less surprising when quickly discovering the API. Thanks for the suggestion though @Abestanis !

**Review by richardeoin — APPROVED** [review]
Thanks for the updates

**Pull request merged**
